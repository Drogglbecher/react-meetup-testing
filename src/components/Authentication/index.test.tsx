import React from 'react';
import AuthenticationContainer from '.';
import {shallow} from 'enzyme';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

describe('ConnectedAuthentication', () => {
  let wrapper;
  let store;
  const mockStore = configureMockStore([thunk]);

  beforeEach(() => {
    const initialState = {authentication: {error: ''}};
    store = mockStore(initialState);
    wrapper = shallow(<AuthenticationContainer store={store} />);
  });

  it('mapStateToProps', () => {
    expect(wrapper.props().error).toEqual('');
  });

  it('mapDispatchToProps with wrong login data', () => {
    // When
    wrapper.simulate('submitCredentials');

    // Then
    const actions = store.getActions();
    expect(actions[0]).toEqual({
      type: '[AUTHENTICATION] SubmitCredentialsStart',
    });
    expect(actions[1]).toEqual({
      payload: 'Invalid authentication data.',
      type: '[AUTHENTICATION] Failed',
    });
  });

  it('mapDispatchToProps with wrong login data [snapshot]', () => {
    // When
    wrapper.simulate('submitCredentials');

    // Then
    const actions = store.getActions();
    expect(actions).toMatchSnapshot();
  });
});
