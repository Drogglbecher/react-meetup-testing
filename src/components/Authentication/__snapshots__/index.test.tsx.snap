// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedAuthentication mapDispatchToProps with wrong login data [snapshot] 1`] = `
Array [
  Object {
    "type": "[AUTHENTICATION] SubmitCredentialsStart",
  },
  Object {
    "payload": "Invalid authentication data.",
    "type": "[AUTHENTICATION] Failed",
  },
]
`;
